interface/src/app/gene/searchMany/searchMany.js

```diff
@@ -36,226 +36,151 @@ angular.module('adage.gene.searchMany', [
-
-              if (results.found.length <= 1) {
-                if (results.found[0]) {
-                  SelectedGenesFactory.addGene(results.found[0]);
-                  SearchResults.remove(key);
-                }
+.directive('searchResultTable', [function() {
+  return {
+    controller: ['$scope', 'SelectedGenesFactory',
+      function($scope, SelectedGenesFactory) {
+        $scope.currentPage = 1;
+        $scope.itemsPerPage = 10;
+        $scope.totalResults = 0;
+        $scope.maxSize = 10;
+        $scope.resultsForPage = [];
+
+        $scope.updatePageGenes = function() {
+          var begin = ($scope.currentPage - 1) * $scope.itemsPerPage;
+          var end = begin + $scope.itemsPerPage;
+          return $scope.queries.slice(begin, end);
+        };
+
+        $scope.addAllNonAmbiguous = function() {
+          // Function to automatically add all genes that
+          // only have one search result.
+          for (var i = 0; i < $scope.queries.length; i++) {
+            var key = $scope.queries[i];
+            var results = $scope.searchResults[key];
+
+            if (results.found.length <= 1) {
+              if (results.found[0]) {
+                SelectedGenesFactory.addGene(results.found[0]);
+                delete $scope.searchResults[key];
               }
             }
-          };
-
-          $scope.removeNotFound = function() {
-            // Function to get rid of all the queries that returned no results.
-            var searchResults = SearchResults.getSearchResults();
-
-            var searchResultKeys = Object.keys(searchResults);
+          }
+        };
 
-            for (var i = 0; i < searchResultKeys.length; i++) {
-              var key = searchResultKeys[i];
-              var results = searchResults[key];
+        $scope.removeNotFound = function() {
+          // Function to get rid of all the queries that returned no results.
+          for (var i = 0; i < $scope.queries.length; i++) {
+            var key = $scope.queries[i];
+            var results = $scope.searchResults[key];
 
-              if (results.found.length === 0) {
-                SearchResults.remove(key);
-              }
+            if (results.found.length === 0) {
+              delete $scope.searchResults[key];
             }
-          };
-        }
-      ],
-
-      link: function(scope, element, attr) {
-        scope.$on('results.update', function() {
-          scope.totalResults = SearchResults.size();
-          scope.resultsForPage = CommonGeneFuncts.updatePageGenes(
-            scope, SearchResults);
-        });
+          }
+        };
 
-        scope.$on('results.loadingSearchResults', function() {
-          scope.loadingSearchResults = true;
+        $scope.$watchCollection('searchResults', function() {
+          $scope.queries = Object.keys($scope.searchResults);
+          $scope.totalResults = $scope.queries.length;
+          $scope.resultsForPage = $scope.updatePageGenes();
```

Comment:
I had forgotten about `$watchCollection`. This seems like a reasonable way to keep this section updated.